.github/workflows/auto-publish.yml

```
@@ -0,0 +1,24 @@
+# Configuration options https://w3c.github.io/spec-prod/
+name: Node CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+jobs:
+  validate-and-publish:
+    name: Validate and Publish
+    runs-on: ubuntu-latest # only linux supported at present
+    steps:
```

To allow writing to `gh-pages`:
```suggestion
    permissions:
      contents: write
    steps:
```